### PR TITLE
Use generated interfaces from eclipse-che/api

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Use interfaces from eclipse-che/api.

### CODE REVIEW

File: extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@
  **********************************************************************/
 
 import { ProxyIdentifier, createProxyIdentifier } from '@theia/plugin-ext/lib/api/rpc-protocol';
-import * as che from '@eclipse-che/plugin';
+import { che as api } from '@eclipse-che/api';
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]: hello, is there a reason to rename che to api ?

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]: There is che in '@eclipse-che/plugin' and che in '@eclipse-che/api'. We have to separate them somehow.

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]: Evhen and me have discussed this and decided that renaming che to api when importing from '@eclipse-che/api' is a variant.

Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]: chePlugin and cheApi ?

IMHO api is too generic when reading the code

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]:
```
import * as chePlugin from '@eclipse-che/plugin';
import { che as cheApi } from '@eclipse-che/api';
```

Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]: maybe it's verbose but it's easy to track in the code (to avoid also mistakes)

Comment (evidolob) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]:
```typescript
import * as chePlugin from '@eclipse-che/plugin';
import { che as cheApi } from '@eclipse-che/api';
```
Is fine for me.

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts]: Reworked

File: extensions/eclipse-che-theia-plugin-ext/src/plugin/che-workspace.ts [hunk 2]
```diff
@@ -10,7 +10,8 @@
 
 import { RPCProtocol } from '@theia/plugin-ext/lib/api/rpc-protocol';
 import { PLUGIN_RPC_CONTEXT, CheWorkspace, CheWorkspaceMain } from '../common/che-protocol';
-import * as che from '@eclipse-che/plugin';
+import { che as api } from '@eclipse-che/api';
+import { KeyValue } from '@eclipse-che/plugin';
```
[the remark this answers is not in the document]
Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/plugin/che-workspace.ts]: Reworked

File: extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts [hunk 3]
```diff
@@ -9,16 +9,10 @@
...
 import { Plugin } from '@theia/plugin-ext/lib/api/plugin-api';
-import {
-    Workspace,
-    WorkspaceConfig,
-    ResourceCreateQueryParams,
-    WorkspaceSettings,
-    Variable,
-    Disposable
-} from '@eclipse-che/plugin';
+import { Variable, Disposable, KeyValue } from '@eclipse-che/plugin';
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts]: we have 
```
import * as che from '@eclipse-che/plugin';
```
and
```
import { Variable, Disposable, KeyValue } from '@eclipse-che/plugin';
```

it looks like the second line of imports should not be used

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts]: Done

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-factory-main.ts [hunk 4]
```diff
@@ -10,7 +10,8 @@
 
 import { interfaces } from 'inversify';
 import { CheFactoryMain } from '../common/che-protocol';
-import { CheApiService, FactoryDto } from '../common/che-protocol';
+import { CheApiService } from '../common/che-protocol';
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-factory-main.ts]:
```
import { CheFactoryMain , CheApiService } from '../common/che-protocol'; 
``` 
?

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/che-factory-main.ts]: Would be nice to have a plugin / extension for che-Theia to auto group imports like that.
Thanks!

**Pull request merged**